src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArr = [];
+  let stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+
+        newArr.push({
+          ...stateClone,
+          ...action.extraData,
+        });
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => delete stateClone[key]);
+        newArr.push({ ...stateClone });
+        break;
+
+      case 'clear':
+        stateClone = {};
+        newArr.push({});
```

Comment:
[Checklist](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) 4th item, you repeat `push` statements 3 times, do it after `switch`